### PR TITLE
Make sure the webview is loaded before sending requests

### PR DESCRIPTION
**Pull request by Or-Geva**
- [ ] All [tests](https://github.com/jfrog/jfrog-vscode-extension#building-and-testing-the-sources) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] I used `npm run format` for formatting the code before submitting the pull request.
-----
Before this pull request, there was a potential for the Webview to take longer to load than the duration between the IDE sending the first request to the Webview. To address this issue, we now ensure that we wait for a signal indicating the completion of the webview's full loading process before sending any requests.

Dependent on: 
* https://github.com/jfrog/jfrog-ide-webview/pull/51

### CODE REVIEW

File: src/main/webview/event/eventManager.ts [hunk 1]
```diff
@@ -5,16 +5,42 @@ import * as vscode from 'vscode';
...
 import { JumpToCodeTask } from './tasks/jumpToCode';
+import { RunUtils } from '../../utils/runUtils';
 
 /**
  * Manages events and communication between the extension and the webview.
  */
 export class EventManager {
     protected send: EventSender;
+    private webviewLoaded: boolean = false;
+    private static TIMEOUT_THREE_MINUTES_IN_MS: number = 3 * 60 * 1000;
```
Comment (yahavi) [inline src/main/webview/event/eventManager.ts:16]: There is no point in mentioning the timeout value in the key
```suggestion
    private static TIMEOUT_MILLISECS: number = 3 * 60 * 1000;
```

File: src/main/webview/event/eventManager.ts [hunk 2]
```diff
@@ -5,16 +5,42 @@ import * as vscode from 'vscode';
...
+    private static RETRY_DELAY_TEN_MS: number = 3 * 60 * 1000;
```
Comment (yahavi) [inline src/main/webview/event/eventManager.ts:17]:
```suggestion
    private static RETRY_DELAY_MILLISECS: number = 3 * 60 * 1000;
```

File: src/main/webview/event/eventManager.ts [hunk 3]
```diff
@@ -5,16 +5,42 @@ import * as vscode from 'vscode';
...
+    ): Promise<EventManager> {
+        const eventManager: EventManager = new EventManager(webview, connectionManager, logManager);
+        await EventManager.waitUntilWebviewLoaded(eventManager);
+        return eventManager;
+    }
+
+    private static async waitUntilWebviewLoaded(eventManager: EventManager) {
+        const startedTime: number = Date.now();
+        for (let i: number = 1; !EventManager.timedOut(startedTime) && !eventManager.webviewLoaded; i++) {
+            eventManager.send.setEventEmitter();
```
Comment (yahavi) [inline src/main/webview/event/eventManager.ts:37]: This method is async. Is it intended to not awaiting?

**Review by yahavi — CHANGES_REQUESTED** [review]
I'm getting an infinite waiting view after clicking on any component:
image

File: src/main/webview/webviewSidebar.ts [hunk 4]
```diff
@@ -41,13 +41,13 @@ export class WebviewSidebar extends WebView implements vscode.WebviewViewProvide
         return currentWebview;
     }
 
-    private createEventManager(webview: vscode.WebviewView) {
-        const eventManager: EventManager = new EventManager(webview.webview, this.connectionManager, this._logManager);
-        webview.onDidChangeVisibility(() => {
+    private async createEventManager(webview: vscode.WebviewView) {
+        const eventManager: EventManager = await EventManager.createEventManager(webview.webview, this.connectionManager, this._logManager);
+        webview.onDidChangeVisibility(async () => {
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply src/main/webview/webviewSidebar.ts:46]: Im not sure im getting your point but we have already performed a cleanup at `currentWebview.onDidDispose` line 28

[the remark this answers is not in the document]
Comment (yahavi) [reply src/main/webview/webviewSidebar.ts:46]: I believe that there is a reason that onDidChangeVisibility returns a disposable. Are you sure that currentWebview.onDidDispose disposes of this listener?

**Review by yahavi — APPROVED** [review]
LGTM

**Pull request merged**